lua/lib/fs.lua

```diff
@@ -241,7 +241,8 @@ end
 function M.rename(node)
   if node.name == '..' then return end
 
-  local new_name = vim.fn.input("Rename " ..node.name.. " to ", node.absolute_path)
+  local abs_path = string.gsub(node.absolute_path, node.name, '')
```

Comment:
I'm not sure if this would work for every case, could you try renaming a file with a `%` or `.` inside ? I'm not too sure but i've encountered this issues with lua regexes a lot, and i believe gsub does apply a regex as 2nd argument, which would lead to undefined behavior or bugs

Reply:
You are correct, I will look into this!